Fix: Window Actions

ibrahimcetin opened the pull request:
<!--- IMPORTANT: If this PR addresses multiple unrelated issues, it will be closed until separated. -->

### Description

This PR fixes #1625. Also, I refactored the code to make it consistent. I removed previous implementation, and refactored it to only use SwiftUI openWindow environment variable.

### Related Issues

<!--- REQUIRED: Tag all related issues (e.g. * #123) -->
<!--- If this PR resolves the issue please specify (e.g. * closes #123) -->
<!--- If this PR addresses multiple issues, these issues must be related to one other -->

* #1625

### Checklist

<!--- Add things that are not yet implemented above -->

- [x] I read and understood the [contributing guide](https://github.com/CodeEditApp/CodeEdit/blob/main/CONTRIBUTING.md) as well as the [code of conduct](https://github.com/CodeEditApp/CodeEdit/blob/main/CODE_OF_CONDUCT.md)
- [x] The issues this PR addresses are related to each other
- [x] My changes generate no new warnings
- [x] My code builds and runs on my machine
- [x] My changes are all related to the related issue above
- [x] I documented my code

<!--- IMPORTANT: Fill out all required fields. Otherwise we might close this PR temporarily -->

austincondiff commented on CodeEdit/Utils/Extensions/NSApplication/NSApp+openWindow.swift:
```diff
@@ -6,30 +6,15 @@
-            NSApp.sendAction(action, to: item.representedObject, from: nil)
-        }
+extension OpenWindowAction {
+    func callAsFunction(sceneID: SceneID) {
```
Can or should we do something like this in order to avoid changing all of the other uses of `openWindow` to include `sceneID`?

```suggestion
    func callAsFunction(_ sceneID: SceneID) {
```

ibrahimcetin replied:
If we omit the parameter name, we still have to change other uses because the previous code uses `openWindow(id:)`. We can overload it like `func callAsFunction(id sceneID: SceneID)` but I don't prefer that because `sceneID` is more descriptive.

austincondiff reviewed (dismissed):
Thanks for fixing this!

FastestMolasses reviewed (dismissed):
LGTM great changes

ibrahimcetin commented:
I just realized I forgot to replace the hard-coded 'settings' string with the SceneID.

matthijseikelenboom commented:
Sorry but I'm gonna be that guy and ask the question no one wants to hear probably. Why don't we now have tests so we know that this bug doesn't reoccur in the future?

ibrahimcetin commented:
@matthijseikelenboom You are right, I agree with you in general but not for this PR because I just removed custom (probably a workaround) `openWindow` solution and replaced it with directly SwiftUI environment value. So, there is no custom logic to test. But if you mean UITest, to be honest, I don't know how to test this code.

matthijseikelenboom commented:
Well there is logic. A button press should open a window. The original logic there also wasn't custom, but it apparently didn't work. Since the button press is implemented with an anonymous function, the only way to test it, is by a UI test. 

To test UI, you need to make a XCUIApplication. But how you trigger the menubar command/button, you'd have to figure out, because I don't know it from the top of my head.
